test/functional/mempool_updatefromblock.py

```diff
@@ -97,10 +99,67 @@ def transaction_graph_test(self, size, n_tx_to_mine=None, fee=100_000):
             assert_equal(entry['ancestorcount'], k + 1)
             assert_equal(entry['ancestorsize'], sum(tx_size[0:(k + 1)]))
 
+        self.generate(self.nodes[0], 1)
+        assert_equal(self.nodes[0].getrawmempool(), [])
+        wallet.rescan_utxos()
+
+    def test_max_disconnect_pool_bytes(self):
+        self.log.info('Creating independent transactions to test MAX_DISCONNECTED_TX_POOL_BYTES limit during reorg')
+
+        # Generate coins for the hundreds of transactions we will make
+        parent_target_vsize = 100_000
+        wallet = MiniWallet(self.nodes[0])
+        self.generate(wallet, (MAX_DISCONNECTED_TX_POOL_BYTES // parent_target_vsize) + 100)
+
+        assert_equal(self.nodes[0].getrawmempool(), [])
+
+        # Set up empty fork blocks ahead of time, needs to be longer than full fork made later
+        fork_length = 60
+        fork_blocks = self.generate(self.nodes[0], fork_length)
+        self.nodes[0].invalidateblock(fork_blocks[0])
```

Comment:
Just a comment, I think it is kind of fine to do the reorgs through invalidateblock instead of connecting and disconnecting nodes from each other, but the disconnectpool behaviour here is kind of on the line where I would tend to prefer exercising the actual code through `ActiveBestChainStep`, since the two methods do not share the same object. I think to test the mempool in general it is fine to do, both methods at least use the same object at runtime.

Reply:
did a proper reorg using blocktools to prep the fork.